src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -412,29 +425,37 @@ bool Connection::websocket_closed_handler(bool was_clean, Status status)
         return bool(m_websocket);
     }
 
-    auto&& status_code = status.code();
-    std::error_code error_code{static_cast<int>(status_code), websocket::websocket_close_status_category()};
```

Comment:
the sync_session expects the `websocket_close_status_category`. Why not have a category for websocket error/close codes instead of the dummy category?